File: frontend/src/objects/review/AppealRole.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
export const APPEAL_ROLE = {
  merchant: 'Merchant',
  rider: 'Rider',
} as const

export type AppealRole = (typeof APPEAL_ROLE)[keyof typeof APPEAL_ROLE]

File: backend/src/main/scala/services/review/objects/AppealRole.scala
package services.review.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import system.objects.*

import io.circe.{Decoder, Encoder}

enum AppealRole:
  case Merchant, Rider

object AppealRole:
  private val enumLabel = text("AppealRole")
  def render(value: AppealRole): DisplayText = enumDisplayText(value)
  def parse(value: DisplayText): Option[AppealRole] = parseEnumValue(value, AppealRole.values)
  given Encoder[AppealRole] = enumEncoder
  given Decoder[AppealRole] = enumDecoder(AppealRole.values, enumLabel)
